src/stores/roles/actions.ts

```diff
@@ -0,0 +1,31 @@
+import { doRequest, get, post, put } from '@/services/http'
+import { getRoles } from './getters'
+import type { Role } from '@/domain/Role'
+import { roles } from './state'
+
+function setRoles(newRoles: Role[]): void {
+  roles.value = newRoles
+}
+
+async function fetchRoles(): Promise<void> {
+  if (getRoles?.value?.length > 0) {
+    return
+  }
+
+  const fetchedRoles = await get<Role[]>('/roles')
+
+  setRoles(fetchedRoles)
+}
+
+async function quizResult(quizAnswer: any) {
+  if (quizAnswer) {
+    try {
+      const result = await doRequest('/calculate-score', 'POST', quizAnswer)
```

Comment:
```suggestion
      const result = await post
```

Using post method, you don't need to pass 'POST' in the second parameter